server/Database/database.js

```diff
@@ -0,0 +1,55 @@
+require('dotenv').config();
+const mongoose = require('mongoose');
+const dbUrl = process.env.MONGO_DB_URL;
+const localHost = process.env.LOCAL_HOST_URL;
+
+class Database
+{
+    constructor()
+    {
+        const url = process.env.IN_DEV === 'true'? localHost: dbUrl;
+        const connect = mongoose.connect(url, {
+            useNewUrlParser : true,
+            useUnifiedTopology: true
+        });
+
+        connect.then((result) => {
+            console.log('Mongoose has established a connection')
+        }).catch((error) => {
+            console.log('Error --> ', error)
+        });
+    }
+
+    getInstanceStatus()
+    {
+        if(process.env.NODE_ENV === 'production')
+        {
+            return '__production';
+        }
+        else{
+            return '__staging'
+        }
+    }
+}
+
+class Singleton
+{
+    constructor()
+    {
+        if(!Singleton.instance)
+        {
+            Singleton.instance = new Database();
+        }
+    }
+
+    getInstance()
+    {
+        return Singleton.instance;
+    }
+}
+
+mongoose.model.on('index', function(err){

```

Comment:
What is this?

Reply:
This listens for errors on the [index calls](https://mongoosejs.com/docs/guide.html#indexes) of the models . It is useful for local development though a bit discouraged for production due to performance implications.

Reply:
Ok ok :)